packages/grafana-data/src/types/app.ts

```diff
@@ -21,15 +21,27 @@ export interface AppRootProps<T = KeyValue> {
   onNavChanged: (nav: NavModel) => void;
 }
 
+export interface AppPageProps<T = any, Q = any, M = any> extends GrafanaRouteComponentProps<T, Q> {
+  meta: AppPluginMeta<M>;
+}
+export type AppRouteComponent<T = any> = React.ComponentType<AppPageProps<T>>;
+
+interface AppRouteDescriptor extends Omit<RouteDescriptor, 'component'> {
+  component: AppRouteComponent<any>;
+  isRoot?: boolean;
+}
 export interface AppPluginMeta<T = KeyValue> extends PluginMeta<T> {
   // TODO anything specific to apps?
 }
 
 export class AppPlugin<T = KeyValue> extends GrafanaPlugin<AppPluginMeta<T>> {
-  // Content under: /a/${plugin-id}/*
+  /**
+   * Content under: /a/${plugin-id}/*
+   * @deprecated
+   */
   root?: ComponentClass<AppRootProps<T>>;
   rootNav?: NavModel; // Initial navigation model
-
+  routes: AppRouteDescriptor[] = [];
```

Comment:
why so much control?  just give the app the react router and let them go crazy

I dont see any value in adding our special wrapper around:
```
path: string; // The URL path to this page
query: KeyValue; // The URL query parameters
```
perhaps that in a distant future when we don't use react :grimacing: